### PR TITLE
Shopowners Cancel Orders

### PR DESCRIPTION
**Pull request by nwochaadim**
#### What does this PR do?

It ensures that Shop owners are able to cancel shop orders
#### What are the relevant pivotal tracker stories?

128320495

### CODE REVIEW

**Comment by betkom** [comment]
@andela-anwocha Please fix merge conflicts

File: app/assets/javascripts/shop.coffee [hunk 1]
```diff
@@ -65,6 +65,7 @@ $ ->
         outputErrors(data.responseJSON)
 
   $('#session_password').trigger('focus')
+
```
Comment (betkom) [inline app/assets/javascripts/shop.coffee]: @andela-anwocha remove the two extra spaces

File: app/assets/javascripts/shop.coffee [hunk 2]
```diff
@@ -76,11 +77,11 @@ outputErrors = (data) ->
   Materialize.toast toastContent
 
 
-window.confirmDeleteProduct = (productIndex) ->
+@confirmDeleteProduct = (productIndex) ->
   document.productIndex = productIndex
   $('#confirm-modal').openModal()
 
-window.deleteProduct = ->
+@deleteProduct = ->
```
Comment (betkom) [inline app/assets/javascripts/shop.coffee:82]: cool

File: app/assets/javascripts/shop_orders.coffee [hunk 3]
```diff
@@ -0,0 +1,40 @@
+$ ->
+  $('a.update-btn').click ->
+    json_array = []
+    $('tr').each ->
+      id = $(this).attr('id')
+      json_object = {}
+      json_object["id"] = id
+      json_object["status"] = $(this).find('select').val()
+      json_array.push(json_object) if id?
```
Comment (betkom) [inline app/assets/javascripts/shop_orders.coffee]: @andela-anwocha best to do `if id then json_array.push(json_object)`

File: app/assets/stylesheets/shop.scss [hunk 4]
```diff
@@ -375,16 +376,29 @@ a > h3 {
 
   h4 {
     display: inline;
-    font-size: 20px;
+    font-size: 23px;
```
Comment (betkom) [inline app/assets/stylesheets/shop.scss]: we should create variables for the most used unit values

File: app/views/shops/orders.html.erb [hunk 5]
```diff
@@ -14,8 +14,8 @@
   <div class="col s5 pagination-holder">
     <ul class="tabs no-border-bottom">
       <li class="tab col s3"><a href="" >All (<%= all_orders_count %>)</a></li>
-      <li class="tab col s3"><a  href="?status=completed">Completed (<%= Order.pending(@shop).count %>)</a></li>
-      <li class="tab col s3"><a href="?status=pending" >Pending (<%= Order.completed(@shop).count %>)</a></li>
+      <li class="tab col s3"><a  href="?status=completed">Completed (<%= Order.completed(@shop).count %>)</a></li>
```
Comment (betkom) [inline app/views/shops/orders.html.erb]: @andela-anwocha You should move the logic outside the view, you could make this a variable in the controller method that you can then access in the view. You should read this to get more insight https://www.toptal.com/ruby-on-rails/top-10-mistakes-that-rails-programmers-make and http://brandonhilkert.com/blog/refactoring-logic-from-a-rails-view/

File: app/views/shops/orders.html.erb [hunk 6]
```diff
@@ -14,8 +14,8 @@
...
+      <li class="tab col s3"><a href="?status=pending" >Pending (<%= Order.pending(@shop).count %>)</a></li>
```
Comment (betkom) [inline app/views/shops/orders.html.erb]: ditto

File: spec/factories/orders.rb [hunk 7]
```diff
@@ -6,7 +6,7 @@
     user factory: :regular_user
     address factory: :address
     status "Pending"
-
+    shop nil
```
Comment (betkom) [inline spec/factories/orders.rb]: @andela-anwocha is there any reason why you are setting the shop field to nil here?

**Comment by betkom** [comment]
@andela-anwocha I had some comments

File: app/assets/javascripts/shop_orders.coffee [hunk 8]
```diff
@@ -6,12 +6,12 @@ $ ->
       json_object = {}
       json_object["id"] = id
       json_object["status"] = $(this).find('select').val()
-      json_array.push(json_object) if id?
+      if id?
```
Comment (betkom) [inline app/assets/javascripts/shop_orders.coffee]: @andela-anwocha You should write this on a line like this `if id then json_array.push(json_object)`

File: app/assets/stylesheets/shop.scss [hunk 9]
```diff
@@ -376,7 +376,7 @@ a > h3 {
 
   h4 {
     display: inline;
-    font-size: 23px;
+    font-size: $title-text;
```
Comment (andela-tkomolafe) [inline app/assets/stylesheets/shop.scss]: This variable name should be more intuitive as we would be using it across the app, something like `gm-font-m` and 17px can be `gm-font-sm` etc

File: app/assets/stylesheets/variables.scss [hunk 10]
```diff
@@ -15,6 +15,7 @@ $light-gray: #f5f5f5;
 $grey-border: #efefef;
 $bottom-border: #e8e8e8;
 $section-color: #333;
+$title-text: 23px;
```
Comment (betkom) [inline app/assets/stylesheets/variables.scss]: ditto my comment above

**Pull request merged**